tests/test_cancellations.py

```diff
@@ -0,0 +1,70 @@
+import typing
+from typing import Optional
+
+import anyio
+import pytest
+
+import httpcore
+from httpcore._async.http2 import HTTPConnectionState as HTTP2ConnectionState
+from httpcore._async.http11 import HTTPConnectionState as HTTP11ConnectionState
+from httpcore._synchronization import AsyncSemaphore
+
+
+class SlowStream(httpcore.AsyncNetworkStream):
+    async def write(
+        self, buffer: bytes, timeout: typing.Optional[float] = None
+    ) -> None:
+        await httpcore._backends.auto.AutoBackend().sleep(2)
+
+    async def aclose(self) -> None:
+        ...
+
+
+class SlowBackend(httpcore.AsyncNetworkBackend):
+    async def connect_tcp(
+        self,
+        host: str,
+        port: int,
+        timeout: Optional[float] = None,
+        local_address: Optional[str] = None,
+        socket_options: typing.Optional[typing.Iterable[httpcore.SOCKET_OPTION]] = None,
+    ) -> httpcore.AsyncNetworkStream:
+        return SlowStream()
+
+
+@pytest.mark.anyio
+async def test_async_cancellation():
+    async with httpcore.AsyncConnectionPool(network_backend=SlowBackend()) as pool:
+        with anyio.move_on_after(0.001):
+            await pool.request("GET", "http://example.com")
+        assert not pool.connections
+
+
+@pytest.mark.anyio
+async def test_h11_response_closed():
+    origin = httpcore.Origin(b"https", b"example.com", 443)
+    stream = httpcore.AsyncMockStream([])
+    async with httpcore.AsyncHTTP11Connection(origin, stream) as conn:
+        # mock
+        with anyio.CancelScope() as cancel_scope:
+            cancel_scope.cancel()
+            await conn.request("GET", "https://example.com")
+        assert conn._state == HTTP11ConnectionState.CLOSED
```

Comment:
I assume we want the test to mirror the same style as the `test_async_cancellation` case above.

Something like this?...

```suggestion
    stream = SlowStream()
    async with httpcore.AsyncHTTP11Connection(origin, stream) as conn:
        with anyio.move_on_after(0.001):
            await pool.request("GET", "http://example.com")
        assert conn.is_closed()
```

Reply:
This looks much better, thanks